qt/python/mantidqt/mantidqt/widgets/sliceviewer/test/test_sliceviewer_presenter.py

```diff
@@ -670,6 +672,110 @@ def run_requests_hkl_test(pres, ncols, callcount):
         pres4D = SliceViewer(mock.Mock(), model=self.model, view=self.view4D)
         run_requests_hkl_test(pres4D, 3, 1)
 
+    @mock.patch("mantidqt.widgets.sliceviewer.presenters.presenter.SliceViewXAxisEditor")
+    def test_x_axes_editor_is_opened_on_x_axes_double_click(self, mock_xaxis_editor):
+        presenter = SliceViewer(None, model=self.model, view=self.view)
+        # mouse event
+        event = mock.MagicMock()
```

Comment:
Thanks for adding the tests - looks very thorough! My one comment is that there is a lot of duplicated setup code for these new tests - for example setting up the mock event could be put in a helper function, likewise a helper function for setting up the data_view (possibly requiring an argument to specify the axis to be clicked so that the correct return values can be set).